src/Sarif/EnumeratedArtifact.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public class EnumeratedArtifact : IEnumeratedArtifact
```

Comment:
>EnumeratedArtifact [](http://example.com/codeflow?start=17&length=18)

EnumeratedArtifact is a thing.

It may exist only as a URI reference, in which case the URI can be resolved to retrieve it.

It can be populated with a stream, in which case there's no need to retrieve/download/load it.

It can be pre-populated with a string representation. We allow this because in the majority of cases we are in the business of analyzing strings. We might need to reconsider this approach, though, as other scan targets won't always be a string. 

Probably, this type needs to be generic, with EnumeratedArtifact<string> ultimately resolving to a string.